toolkit/tools/internal/file/file.go

```diff
@@ -75,13 +86,19 @@ func Move(src, dst string) (err error) {
 // Copy copies a file from src to dst, creating directories for the destination if needed.
 // dst is assumed to be a file and not a directory. Will preserve permissions.
 func Copy(src, dst string) (err error) {
-	return copyWithPermissions(src, dst, os.ModePerm, false, os.ModePerm)
+	return copyWithPermissions(src, dst, os.ModePerm, false, os.ModePerm, false)
 }
 
 // CopyAndChangeMode copies a file from src to dst, creating directories with the given access rights for the destination if needed.
 // dst is assumed to be a file and not a directory. Will change the permissions to the given value.
 func CopyAndChangeMode(src, dst string, dirmode os.FileMode, filemode os.FileMode) (err error) {
-	return copyWithPermissions(src, dst, dirmode, true, filemode)
+	return copyWithPermissions(src, dst, dirmode, true, filemode, false)
+}
+
+// Copy copies a file from src to dst, creating directories for the destination if needed.
+// dst is assumed to be a file and not a directory. Will preserve permissions and symlinks.
+func CopyNoDereference(src, dst string) (err error) {
```

Comment:
random thought, probably an overkill: with so many versions of func copy*, do you think having a builder would be more appropriate?

Reply:
Fixed